### DISCUSSION TITLE
how to solve video player issue

### DISCUSSION THREAD

**Discussion by arupbkp** [ant-media/Ant-Media-Server#6157] [General]
<img width="1328" alt="Screenshot 2024-03-05 at 9 33 02 AM" src="https://github.com/ant-media/Ant-Media-Server/assets/116704492/e85ad470-aece-4607-94e1-53e3f163b27b">

E/flutter ( 9949): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type '_Map<String, dynamic>' is not a subtype of type 'String' of 'other'

**Comment by yashtandon113** [comment]
@arupbkp 
Thanks for reporting the issue.

Unfortunately, the issue is not very clear. Could you please share detailed information about what you are trying to achieve? Also, it would be much appreciated if you could share the step-by-step reproduce scenario to replicate the issue.

**Reply by arupbkp** [reply]
Video player in a list view returning multiple errors and app hang.
Please help me.

E/flutter ( 7408): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type '_Map<String, dynamic>' is not a subtype of type 'String' of 'other'
E/flutter ( 7408): #0      AntHelper.onMessage (package:ant_media_flutter/src/helpers/helper.dart:277:30)
E/flutter ( 7408): #1      AntHelper.connect.<anonymous closure> (package:ant_media_flutter/src/helpers/helper.dart:364:12)
E/flutter ( 7408): #2      SimpleWebSocket.connect.<anonymous closure> (package:ant_media_flutter/src/utils/websocket.dart:27:26)
E/flutter ( 7408): #3      _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
E/flutter ( 7408): #4      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
E/flutter ( 7408): #5      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter ( 7408): #6      _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:784:19)
E/flutter ( 7408): #7      _StreamController._add (dart:async/stream_controller.dart:658:7)
E/flutter ( 7408): #8      _StreamController.add (dart:async/stream_controller.dart:606:5)
E/flutter ( 7408): #9      new _WebSocketImpl._fromSocket.<anonymous closure> (dart:_http/websocket_impl.dart:1144:21)
E/flutter ( 7408): #10     _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
E/flutter ( 7408): #11     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
E/flutter ( 7408): #12     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter ( 7408): #13     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:63:11)
E/flutter ( 7408): #14     _EventSinkWrapper.add (dart:async/stream_transformers.dart:13:11)
E/flutter ( 7408): #15     _WebSocketProtocolTransformer._messageFrameEnd (dart:_http/websocket_impl.dart:332:23)
E/flutter ( 7408): #16     _WebSocketProtocolTransformer.add (dart:_http/websocket_impl.dart:226:46)
E/flutter ( 7408): #17     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:111:24)
E/flutter ( 7408): #18     _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
E/flutter ( 7408): #19     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
E/flutter ( 7408): #20     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter ( 7408): #21     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:784:19)
E/flutter ( 7408): #22     _StreamController._add (dart:async/stream_controller.dart:658:7)
E/flutter ( 7408): #23     _StreamController.add (dart:async/stream_controller.dart:606:5)
E/flutter ( 7408): #24     _Socket._onData (dart:io-patch/socket_patch.dart:2447:41)
E/flutter ( 7408): #25     _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
E/flutter ( 7408): #26     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
E/flutter ( 7408): #27     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter ( 7408): #28     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:784:19)
E/flutter ( 7408): #29     _StreamController._add (dart:async/stream_controller.dart:658:7)
E/flutter ( 7408): #30     _StreamController.add (dart:async/stream_controller.dart:606:5)
E/flutter ( 7408): #31     _RawSecureSocket._sendReadEvent (dart:io/secure_socket.dart:1116:19)
E/flutter ( 7408): #32     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:18:15)
E/flutter ( 7408): #33     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:398:19)
E/flutter ( 7408): #34     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
E/flutter ( 7408): #35     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)

**Reply by yashtandon113** [reply]
@arupbkp 
Do you mean you are facing issues in the Flutter SDK during the playback with the play sample? If yes, could you please share the step-by-step reproduce scenario so we can check the issue on our end by replicating it?

It would be great if you could share the details in the below template.

### Short description
__Brief description of what happened__


### Environment

* Operating system and version:
* Java version: 
* Ant Media Server version:
* Browser name and version:

### Steps to reproduce
1. 
2. 
3.

### Expected behavior
Put as much detail here as possible.

### Actual behavior
Put as much detail here as possible.

**Comment by arupbkp** [comment]
How to create a proper live streaming video player and share code.

**Reply by USAMAWIZARD** [reply]
Hi @arupbkp ,
[here](https://github.com/ant-media/WebRTC-Flutter-SDK/tree/main/example/Play) is the example for playing the stream with ant media
